applications/Bifrost-PLO-Liquidity.md

```diff
@@ -0,0 +1,165 @@
+- [Edwin](https://github.com/ark930)
+- [Buffalo](https://github.com/lvqinghao)
+- [Jamie](https://github.com/Dengjianping)
+- [Herry](https://github.com/herryho)
+- [Carmen](https://github.com/carmencitaqiu)
+
+## Technical solutions
+In this solution, Bifrost provides an intermediate abstraction layer between investors, parachain projects, and the Polkadot relay chain, providing investors with better financial tools and flexibility, and also providing parachain projects Standardized and configurable PLO tools help parachain projects to better crownloan for parachain slots and enrich the ecosystem of the entire Polkadot community.
+
+- Investers: DOT, KSM token holders, parachain users and investment institutions are all potential investors in the parachain slot auction.
+- Parachain: Need to auction parachain slots, access polkadot or kusama relay projects.
+- Bifrost: Through decentralized provision of vsDOT or vsKSM derivatives, the middle layer protocol that unlocks the liquidity of the parachain slot auction。
+- Polkadot/Kusama: Relay chain that has parachain slots and released for auction winners.
+
+Bifrost team will communicate and cooperate with potential parachain projects. Their Investors will transfer DOT/KSM to the Bifrost parachain and send transactions to call the mint-function integrated in Bifrost-Runtime to generate vsDOT/vsKSM. Bifrost's design goal is to become a parachain of Polkadot and/or Kusama. Bifrost collects investors' DOT/KSM and votes on other parachain candidates on behalf of investors. Bifrost Team does not have the rights and cannot control the DOT/KSM from investor users, and these funds will be automatically/decentralizedly kept and controlled by Bifrost-Parachain's code.
+
+The Bifrost Parachain can be treated as an ordinary investor, just like Alice, Bob, Bonding a part of the funds (DOT/KSM), and voting on Parachain A, Parachain B... From this point of view, Bifrost is only a PLO Contributor for other Parachain projects, but Bifrost's funds are from other investors.
+
+And how we make vsToken fungible under different unbonding periods? During vsToken minting, vsDOT/vsKSM and Expire-Date-For-ParachainXXX-Token (the certification of a parachain's contribution) will be generated at the same time. vsDOT is fungible, and vsKSM is also fungible, but the Expire-Date-For-ParachainXXX-Token and Expire-Date-For-ParachainYYY-Token corresponding to different periods and different parachains are different, and not fungible with each other. When the ParachainA slot expires, the relay chain will return part of the DOT/KSM to Bifrost, and any user holding vsDOT/vsKSM and Expire-Date-For-ParachainA-Token will be able to redeem the corresponding DOT/KSM. If a user only has vsDOT/vsKSM, but does not have Expire-Date-For-ParachainA-Token, he will not be able to perform the redemption operation, but he can choose to sell vsDOT/vsKSM, or buy Expire-Date-For-ParachainA-Token to perform the redemption operation.
```

Comment:
Thank you for the clarifications. This means vTokens only "appear" to be fungible but essentially a vToken is worthless without a certificate and vice-versa. This bears some more questions.

- What happens to the underlying Asset if certificates OR vTokens are lost?
- Will there be a market-place to trade those certificates? If yes, I can imagine that liquidity will be a big issue when there are many non-fungible certificates around.
- What if Parachains want to gradually pay out rewards during the time of their operation. Where should they allocate the rewards to, vTokens or certificate?

I can imagine that this will be quite painful from a UX-perspective. What are the plans on that front?

Reply:
Hi Jonas, thanks for your valuable questions, we've added a part of Project Architecture to explain your questions seriously, and pls be aware of we've changed the name of "certificate" to "vsBond", pls feel free to check it.

Reply:
If there is still any question, pls don't hesitate to ask.

Reply:
Thank you for clarifications in the main document. However, I have still a few additional questions:

Do I understand correctly, that you are aiming for three markets? One exchange pool where users can claim tokens against vsToken and expired vsBond, one Bancor Pool to trade vsToken/Token and a third with and a "buy-in-form" for vsBond/Token?
  - How does that buy-in-form look like, what do you mean by it? Is it a fixed price? How is a buyer matched with a seller?
  - How do you handle that those tokens are not fungible at all? Not only are they not fungible in terms of their expiry date but also in terms of the fact that the certificates received different rewards from different parachains. 

You want to fill the bancor pool after a parachain retires with 5% of the unclaimed tokens per day and thereby you are the liquidity prover? 
- This could cause liquidity problems of the exchange pool and they might not be able to redeem their Tokens by vsToken and vsBond? They might be forced to trade separately on the other pools. This will lead to the fact that users who hold vsToken and vsBonds will get less than 1:1 ratio of their underlying asset.

I am also still concerned of the fact that one market side can hold the other hostage. Even though the bancor pool will allow to redeem Token by only providing vsToken, a hostage situation where vsBonds are hoarded would drive up the demand for Tokens in the bancor pool which would be reflected in the exchange ratio.

Reply:
Thank you for your questions. Yes, Bifrost has 3 pools. The first pool is a 1:1 exchange pool, the second is the Bancor pool, and the third pool is the entire external market (ex, dex).
In the first pool, as long as the amount of Token is greater than 0, users can exchange Token back at a price of 1:1 using vsToken+vsBound.  In this period of time, the system will gradually withdraw a portion (5%) of the funds every day and put it into the Bancor-Pool to prevent vsBond from becoming a hostage. If some users only have vsToken, they can still get Token back through Bancor-Pool. After the slot expires, the relaychain will return token to Bifrost. As long as the holder of the vsToken waits, they will be able to exchange the token back at price of 1:1, and the vsBond will not remain the hostage. Through these three methods, Bifrost provides liquidity for tokens invested in PLO.

Reply:
We can set a parameter, such as 4 months, after the slot has expired 4 months, the corresponding vsBond will be invalid, which can also prevent vsBond from becoming a hostage.

Reply:
Thank you for answering my questions.

Reply:
Thank you for your questions, how can we access this merging?